lib/src/component2_suggestors/defaultprops_initialstate_migrator.dart

```diff
@@ -0,0 +1,168 @@
+// Copyright 2019 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:codemod/codemod.dart';
+import 'package:over_react_codemod/src/component2_suggestors/component2_utilities.dart';
+
+/// Abstract suggestor with visitor that can migrate `getDefaultProps`
+/// or `getInitialState`.
+///
+/// Use [GetDefaultPropsMigrator] or [GetInitialStateMigrator].
+abstract class DefaultPropsInitialStateMigrator extends GeneralizingAstVisitor
+    with AstVisitingSuggestorMixin
+    implements Suggestor {
+  bool allowPartialUpgrades = true;
+  bool shouldUpgradeAbstractComponents = false;
+
+  // Can be either `getDefaultProps` or `getInitialState`.
+  String methodToMigrate = 'getDefaultProps';
+  String migrateTo = 'defaultProps';
+
+  @override
+  visitMethodDeclaration(MethodDeclaration node) {
+    super.visitMethodDeclaration(node);
+
+    var containingClass = node.parent;
+
+    if ((!allowPartialUpgrades &&
+            !fullyUpgradableToComponent2(containingClass)) ||
+        (!shouldUpgradeAbstractComponents &&
+            canBeExtendedFrom(containingClass))) {
+      return;
+    }
+
+    if (extendsComponent2(containingClass) &&
+        node.name.name == methodToMigrate) {
+      // Remove return type.
+      if (node.returnType != null) {
+        yieldPatch(
+          node.returnType.offset,
+          node.returnType.end,
+          '',
+        );
```

Comment:
Yes there does end up being an extra space before the getter and the line you mentioned in your other comment is not indented correctly either, but after running the dart formatter both of these are fixed. Are you wanting the codemod to fix these spacing issues itself so that the dart formatter doesn't need to be run?

Reply:
Let's let the formatter handle this. Many of our other codemods already don't output nice whitespace, so they need to format after the codemod either way.